generics/services/sunbird.js

```diff
@@ -0,0 +1,172 @@
+        if (requestType == "PATCH") {
+            request.patch(url, options, callback);
+        } else if (requestType == "GET") {
+            request.get(url, options, callback);
+        } else {
+            request.post(url, options, callback);
+        }
+
+        function callback(err, data) {
+
+            if (err) {
+                 return reject({
+                    message: CONSTANTS.apiResponses.SUNBIRD_SERVICE_DOWN
+                });
+            } else {
+                return resolve(data.body);
+            }
+        }
+
+    });
+}
+
+
+/**
+* To get list of learning resources
+* @method
+* @name  learningResources
+* @param {String} token - user access token.
+* @param {String} pageSize - page size of the request
+* @param {String} pageNo - page no of the request
+* @param {String} category - category for the learning resource
+* @param {String} subCategory - subcategory for the learning resource
+* @param {String} topic -  topic for the learning resource
+* @param {String} language - language for the learning resource
+* @param {String} sortBy - sortBy option for the learning resource
+* @returns {json} Response consists of list of learning resources
+*/
+
+const learningResources = function (token, pageSize, pageNo, filters, sortBy) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            let learningResourceApiUrl = CONSTANTS.endpoints.GET_RESOURCES_LIST
+            learningResourceApiUrl = learningResourceApiUrl + "?limit=" + pageSize + "&page=" + pageNo + "&sortBy=" + sortBy;
+            let mappedFilterList = {};
+            let filterKeys = Object.keys(filters);
+            
+            if (filterKeys && filterKeys.length > 0) {
+                filterKeys.map(filter => {
+                    let mappingType = "";
+                   
+                    if (filter == "category") {
+                        mappingType = "board"
+                    } else if (filter == "subCategory") {
+                        mappingType = "gradeLevel"
+                    } else if (filter == "topic") {
+                        mappingType = "medium"
+                    } else if (filter == "language") {
+                        mappingType = "subject"
+                    }
+                    mappedFilterList[mappingType] = filters[filter];
+                });
+            }
+
+            let requestBody = {
+                filters: mappedFilterList
+            }
+            
+            let response = await callToSunbird("POST", learningResourceApiUrl, token, requestBody);
+            return resolve(response);
+        } catch (error) {
+            reject({ message: CONSTANTS.apiResponses.SUNBIRD_SERVICE_DOWN });
+        }
+
+
+    })
+}
+
+/**
+  * Get filters of learning resources
+  * @function
+  * @name filtersList
+  * @param token - Logged in user token.
+  * @returns {JSON} - consist filters list of learning resources
+*/
+const filtersList = function (token) {
```

Comment:
@rakeshSgr - All APIs of sunbird service have to work only with internal access token